engine/orchestration/src/com/cloud/agent/manager/AgentManagerImpl.java

```diff
@@ -1049,7 +1044,13 @@ public boolean executeUserRequest(final long hostId, final Event event) throws A
             }
             return true;
         } else if (event == Event.ShutdownRequested) {
-            return reconnect(hostId);
+            //should throw a exception here as well.instead of eating this up.
+           try {
+               reconnect(hostId);
+           } catch (CloudRuntimeException e) {
```

Comment:
@sureshanaparti 
I agree with you, I have also added a comment in the code indicating the same. We can work on it in a different pr. This is just a small improvement to the existing code. Fixing this will need lot of rework and cause code churn. So the intention was to do this later in a separate PR.